File: app/src/main/java/com/bureng/robocoinx/presenter/SplashPresenter.java
package com.bureng.robocoinx.presenter;

import android.content.Context;

import com.bureng.robocoinx.contract.SplashContract;
import com.bureng.robocoinx.utils.RoboHandler;
import com.bureng.robocoinx.model.common.UserCache;
import com.bureng.robocoinx.utils.CacheContext;
import com.bureng.robocoinx.utils.StaticValues;

public class SplashPresenter implements SplashContract.Presenter {

    private Context ctx;
    private SplashContract.View view;

    public SplashPresenter(Context _ctx, SplashContract.View _view){
        ctx = _ctx;
        view = _view;
    }

    @Override
    public boolean authorize() {
        UserCache userCache = new CacheContext<>(UserCache.class, ctx).get(StaticValues.USER_CACHE);
        if(userCache == null)
            return false;
        else
            return true;
    }

    @Override
    public Object getResponse() {
        if(authorize())
            return RoboHandler.parsingHomeResponse(ctx);
        else
            return RoboHandler.getSignUpRequest(ctx);
    }
}

File: app/src/main/java/com/bureng/robocoinx/presenter/SplashPresenter.kt
package com.bureng.robocoinx.presenter

import android.content.Context
import com.bureng.robocoinx.contract.SplashContract
import com.bureng.robocoinx.model.common.UserCache
import com.bureng.robocoinx.model.request.SignUpRequest
import com.bureng.robocoinx.model.view.ProfileView
import com.bureng.robocoinx.utils.CacheContext
import com.bureng.robocoinx.utils.RoboHandler
import com.bureng.robocoinx.utils.StaticValues

class SplashPresenter(private val ctx: Context, private val view: SplashContract.View) : SplashContract.Presenter {
    override fun authorize(): Boolean {
        val userCache = CacheContext(UserCache::class.java, ctx)[StaticValues.USER_CACHE]
        return userCache != null
    }
    override fun loadActivity() {
        if (authorize()){
            val resp = RoboHandler.parsingHomeResponse(ctx)
            view.goHome(resp as ProfileView)
        }else{
            val resp = RoboHandler.getSignUpRequest(ctx)
            view.loadFingerprint(resp as SignUpRequest)
        }
    }
}